File: leetcode_ly/java/String/LongestCommonPrefix.java
public class Solution {
    public String longestCommonPrefix(String[] strs) {
        if(strs.length == 0) return "";
        if(strs.length == 1) return strs[0];
        int len = strs[0].length();
        for(int i = 1; i < strs.length; i++){
            if(strs[i].length() < len) len = strs[i].length();
        }
        
        for(int i = len; i > 0; i--){
            int j = 1;
            for(; j < strs.length; j++){
                if(!strs[j].substring(0,i).equals(strs[j - 1].substring(0,i))) break;
                else continue;
            }
            if(j == strs.length) return strs[j - 1].substring(0,i);
        }
        return "";
    }
}

File: leetcode_ly/c++/String/LongestCommonPrefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(strs.size() == 0) return "";
        int i;
        string res = strs[0];
        for(i = 1; i < strs.size(); i++){
            res = compare(res,strs[i]);
        }
        return res;
    }
    
    string compare(string a, string b){
        int j = 0;
        string s = "";
        int lenA = a.size();
        int lenB = b.size();
        while(j < lenA && j < lenB){
            if(a[j] == b[j]){
                s += a[j];
                j++;
            }
            else break;
        }
        return s;
    }
};